variant-inc/actions-cpp
# [CLOUD-1365] actions-cpp

# Description

Creating actions-cpp to build C++ images with conan and push to jfrog repository. Actions-cpp is also capable of pushing docker images to AWS ECR but the C++ repositories are not ready for that yet (`prime`).

Things to check for:
- Dockerfile
- Entrypoint.sh
- Action.yml
- Readme.MD (MOST IMPORTANT)

Fixes [#1365](https://drivevariant.atlassian.net/browse/CLOUD-1365)

## Type of change



## How Has This Been Tested?


  - https://github.com/variant-inc/mirage/actions/runs/1723965692

```yaml
Test Steps:
   1.  Navigate to https://github.com/variant-inc/mirage/tree/feature/CLOUD-1365-actions-cpp
   2.  Create a dummy commit on this branch with the current yaml.
         - name: Actions Cpp
            id: actions-cpp
            uses: variant-inc/actions-cpp@feature/CLOUD-1365-create-actions-c++
            env:
              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
            with:
              conan_push_enabled: true
              env_cyberton: https://drivevariant.jfrog.io/artifactory/api/conan/cybertron-conan
              conan_key: ${{secrets.CONAN_KEY}}
              env_conan_username: rajeevc
              gcc_version: 9
    3. After the action has passed, navigate to | https://drivevariant.jfrog.io/ui/login/
    4. Login with credentials from here | https://drivevariant.atlassian.net/wiki/spaces/CLOUD/pages/2194047011/Credentials
    5. Navigate to the left nav bar, next to the "gear icon", select the "4 squares icon" for Applications
    6. Select Artifactory > Artifacts
    7. Select "cyberton-conan-local">"feature">"mirage">{BUILD_NUMBER}>"feature-CLOUD-1365-actions-cpp"
    8. Select the package_number
    9. Select 'Conan Info' on the right
    
Results:
   1. Verify Below:
   
    Name: | mirage
    -- | --
    Version: | {BUILD_NUMBER}
    User: | feature
    Channel: | feature-CLOUD-1365-actions-cpp
    Reference: | mirage/162@feature/feature-CLOUD-1365-actions-cpp
    License: | Variant

```
## Checklist
